qc-runner/src/init.r

```diff
@@ -214,4 +217,37 @@ wrapper <- function(input_json) {
     return(message_id)
 }
 
-message("New wrapper loaded.")
+init <- function() {
+    pipeline_config <- load_config('host.docker.internal')
+    states <- paws::sfn(config=pipeline_config$aws_config)
+
+    repeat {
```

Comment:
What's our plan to cleanup pipeline servers?

Reply:
The plan is to simply quit the server if there was no message received by it. GetActivityTask [polls for a minute](https://docs.aws.amazon.com/step-functions/latest/apireference/API_GetActivityTask.html) for new work, and if it can't find anything it will return an [empty string](https://paws-r.github.io/docs/sfn/get_activity_task.html) in R in particular.

There is a way to do more advanced error handling and recovery, but because we are not persisting intermediate count matrices to S3, it is probably better to just leave it as is and just quit successfully if we get an empty task token. Mind you, right now with the `tryCatch` block if there is an error being thrown in the step it gets automatically written off as a failure as we catch exceptions and immediately return a failed activity. If there's a condition where we'd end up with no task getting to the worker in a minute it probably makes sense to just nuke the whole thing anyways.

Reply:
Makes sense, thanks. I tend to forget that every pipeline execution starts everything from scratch: new state machine, new state machine execution, new runner job...

BTW: _new runner job_ makes me think that, after remoter client/server setup has been gone, we don't need to have a service as a part the HelmRelease

Reply:
That would be correct, yes. I'll remove it from the chart just now.